app/src/sharedTest/java/org/oppia/android/app/devoptions/ViewEventLogsActivityTest.kt

```diff
@@ -0,0 +1,644 @@
+      createViewEventLogsActivityIntent()
+    ).use {
+      testCoroutineDispatchers.runCurrent()
+      onView(withId(R.id.view_event_logs_recycler_view)).check(hasItemCount(5))
+    }
+  }
+
+  @Test
+  fun testViewEventLogsActivity_checkRecentLogsAreDisplayedFirst() {
+    launch<ViewEventLogsActivity>(
+      createViewEventLogsActivityIntent()
+    ).use {
+      testCoroutineDispatchers.runCurrent()
+      scrollToPosition(0)
+      verifyTextOnEventLogItemViewAtPosition(
+        position = 0,
+        stringToMatch = "Open Revision Card",
+        targetViewId = R.id.view_event_logs_action_name_text_view
+      )
+      scrollToPosition(1)
+      verifyTextOnEventLogItemViewAtPosition(
+        position = 1,
+        stringToMatch = "Open Story Activity",
+        targetViewId = R.id.view_event_logs_action_name_text_view
+      )
+      scrollToPosition(2)
+      verifyTextOnEventLogItemViewAtPosition(
+        position = 2,
+        stringToMatch = "Open Lessons Tab",
+        targetViewId = R.id.view_event_logs_action_name_text_view
+      )
+      scrollToPosition(3)
+      verifyTextOnEventLogItemViewAtPosition(
+        position = 3,
+        stringToMatch = "Open Home",
+        targetViewId = R.id.view_event_logs_action_name_text_view
+      )
+      scrollToPosition(4)
+      verifyTextOnEventLogItemViewAtPosition(
+        position = 4,
+        stringToMatch = "Open Profile Chooser",
+        targetViewId = R.id.view_event_logs_action_name_text_view
+      )
+    }
+  }
+
+  @Test
+  fun testViewEventLogsActivity_configChange_checkRecentLogsAreDisplayedFirst() {
+    launch<ViewEventLogsActivity>(
+      createViewEventLogsActivityIntent()
+    ).use {
+      testCoroutineDispatchers.runCurrent()
+      onView(isRoot()).perform(orientationLandscape())
+      scrollToPosition(0)
+      verifyTextOnEventLogItemViewAtPosition(
+        position = 0,
+        stringToMatch = "Open Revision Card",
+        targetViewId = R.id.view_event_logs_action_name_text_view
+      )
+      scrollToPosition(1)
+      verifyTextOnEventLogItemViewAtPosition(
+        position = 1,
+        stringToMatch = "Open Story Activity",
+        targetViewId = R.id.view_event_logs_action_name_text_view
+      )
+      scrollToPosition(2)
+      verifyTextOnEventLogItemViewAtPosition(
+        position = 2,
+        stringToMatch = "Open Lessons Tab",
+        targetViewId = R.id.view_event_logs_action_name_text_view
+      )
+      scrollToPosition(3)
+      verifyTextOnEventLogItemViewAtPosition(
+        position = 3,
+        stringToMatch = "Open Home",
+        targetViewId = R.id.view_event_logs_action_name_text_view
+      )
+      scrollToPosition(4)
+      verifyTextOnEventLogItemViewAtPosition(
+        position = 4,
+        stringToMatch = "Open Profile Chooser",
+        targetViewId = R.id.view_event_logs_action_name_text_view
+      )
+    }
+  }
+
+  @Test
+  fun testViewEventLogsActivity_contextIsNull_isNotDisplayed() {
```

Comment:
Correct me if I am wrong here, 
in condition `contextIsNull` no logs will be displayed, right?

If yes, we can rename the test as `testViewEventLogsActivity_contextIsNull_logsAreNotDisplayed`

Reply:
Ah. Actually, when the event context is null then the **context field** is not displayed. The name of the may have been confusing so I have renamed it to `testViewEventLogsActivity_contextIsNull_contextIsNotDisplayed`. Does this look good?

Reply:
bit confused with `contextIsNotDisplayed`, what does context implies here?

Reply:
`context` here is a property of `EventLog`. You can see below: https://github.com/oppia/oppia-android/blob/a918b1cf905ea27dabe8cb35da825e94cd6d8b7b/model/src/main/proto/oppia_logger.proto#L20-L35

Reply:
ok, let's keep the name intact.

Reply:
@yashraj-01 have you confirmed the non-displaying of context if null from @rt4914?

Reply:
@Sarthak2601 @yashraj-01 If context is not available we can hide that specific textview.

Reply:
Thanks for the clarification @rt4914